### PR TITLE
Remove snapshot id param (aka: goodbye auto-scrolling behavior)

### PR DESCRIPTION
**Pull request by cadeParade**
Removing the snapshot id query parameter solves both scroll-related issues in clubhouse.
https://app.clubhouse.io/percy/story/1594/auto-scrolling-is-broken-in-builds-with-a-mix-of-approved-unapproved-snapshots
https://app.clubhouse.io/percy/story/1599/if-click-to-approve-a-snapshot-then-scroll-down-to-the-next-one-without-waiting-for-the-approval-once-the-first-one-is-approved

Removing this means that clicking in a snapshot will not generate a unique URL, and navigating to a URL with a snapshot id query param will not scroll to that snapshot anymore. But we can still get a unique URL from a snapshot by either using the nice copy-link button or copying the full-screen page url.

The keyboard nav for hopping up and down snapshots still works, including for builds that have more than 150 diffs (where they are all collapsed). Hopping up and down via keyboard still scrolls the page to the correct snapshot.

However, to make the keyboard nav work, there still needs to be some scroll behavior. I've rewritten the scroll behavior to be less code and make more sense (IMO).

This *does not* fix the bug currently in production where, when closing a full screen snapshot, it does _not_ scroll the the correct position for that snapshot.

## QA suggestions:
Click on some snapshots. Does what you expect happen? Click approve, click some more snapshots, scroll and click some snapshots, use the arrow keys to navigate, open the no-diff batch of snapshots and click/keyboard nav around there.

### CODE REVIEW

File: app/components/snapshot-list.js [hunk 1]
```diff
@@ -108,97 +110,21 @@ export default Component.extend({
...
-      let selectedComponent = snapshotComponents.objectAt(selectedIndex);
-      let lastIndex = this.get('lastSnapshotIndex');
-
-      this.get('updateActiveSnapshotId')(selectedComponent.get('snapshot.id'));
-
-      // Expand the selected component.
-      selectedComponent.get('setAsSelected').call(selectedComponent);
+  _snapshotIds: mapBy('computedSnapshots', 'id'),
+  _numSnapshots: alias('computedSnapshots.length'),
+  _calculateNewActiveSnapshotId({isNext = true} = {}) {
```
Comment (fotinakis) [inline app/components/snapshot-list.js]: you'll have to explain this syntax to me sometime ;)

File: app/components/snapshot-viewer.js [hunk 2]
```diff
@@ -31,57 +29,41 @@ export default Component.extend({
...
-  isDefaultExpanded: true,
-  isFocus: false,
-  isExpanded: computed('isDefaultExpanded', 'snapshot.isApproved', 'build.isApproved', function() {
-    if (this.get('build.isApproved')) {
-      return true;
-    } else if (this.get('snapshot.isApproved')) {
-      return false;
-    } else {
-      return this.get('isDefaultExpanded');
+  isActiveSnapshot: computed('activeSnapshotId', 'snapshot.id', function() {
```
Comment (fotinakis) [inline app/components/snapshot-viewer.js]: So every snapshot-viewer computes if it is the current active snapshot? I guess that's probably fine, but probably not super performant in some situations.

File: tests/integration/components/snapshot-list-test.js [hunk 3]
```diff
@@ -115,4 +113,57 @@ describe('Integration: SnapshotList', function() {
     expect(SnapshotList.isNoDiffsBatchVisible).to.equal(false);
     expect(SnapshotList.snapshots().count).to.equal(numSnapshots);
   });
+
+  describe('when there are more than 150 snapshots with diffs', function() {
```
Comment (fotinakis) [inline tests/integration/components/snapshot-list-test.js]: Yay thank you! 🎉

File: tests/integration/components/snapshot-list-test.js [hunk 4]
```diff
@@ -115,4 +113,57 @@ describe('Integration: SnapshotList', function() {
...
+      });
+      percySnapshot(this.test);
+    });
+
+    it('allows keyboard nav with up and down arrows', function() {
+      SnapshotList.typeDownArrow();
+      expect(SnapshotList.snapshots(0).isExpanded).to.equal(true);
+      expect(SnapshotList.snapshots(0).isFocused).to.equal(true);
+      expect(SnapshotList.snapshots(1).isExpanded).to.equal(false);
+      expect(SnapshotList.snapshots(1).isFocused).to.equal(false);
```
Comment (fotinakis) [inline tests/integration/components/snapshot-list-test.js]: Take a screenshot to see the expanded one, even though the scroll position will be wrong?

File: tests/pages/build-page.js [hunk 5]
```diff
@@ -51,12 +48,8 @@ const BuildPage = {
     return `/${build.project.fullSlug}/builds/${build.id}?snapshot=${snapshot.id}`;
   },
 
-  typeDownArrow: triggerable('keydown', SELECTORS.SNAPSHOT_LIST, {
-    eventProperties: {keyCode: DOWN_ARROW_KEY},
-  }),
-  typeUpArrow: triggerable('keydown', SELECTORS.SNAPSHOT_LIST, {
-    eventProperties: {keyCode: UP_ARROW_KEY},
-  }),
+  typeDownArrow: alias('snapshotList.typeDownArrow'),
```
Comment (fotinakis) [inline tests/pages/build-page.js]: I learn something new Ember from you every time I read your code ⚡️  this is a nice way to keep things DRY.

File: app/components/snapshot-list.js (hunk 1 above)
Comment (cadeParade) [reply app/components/snapshot-list.js]: The syntax is super weird. But this is es6's attempt at named arguments with defaults: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment and cmd+f for "Setting a function parameter's default value"

File: tests/pages/build-page.js (hunk 5 above)
Comment (cadeParade) [reply tests/pages/build-page.js]: Anna A contributed this property to ember page objects :)

**Comment by djones** [comment]
Should we remove the subtle border darken when you select a snapshot?

screen shot 2018-02-26 at 5 30 36 pm

**Comment by timhaines** [comment]
I think the only thing the border is helpful for now, is knowing which snapshot you have selected before pressing an up/down key.  It doesn't impact other interactions/decisions?

**Comment by cadeParade** [comment]
That's right @timhaines it no longer does anything except mark which snapshot your keyboard nav will start from.

**Pull request merged**